picus/src/expr/impls.rs

```diff
@@ -494,3 +494,111 @@ impl GetExprHash for NotExpr {
 }
 
 impl ExprLike for NotExpr {}
+
+//===----------------------------------------------------------------------===//
+// DetExpr
+//===----------------------------------------------------------------------===//
+
+#[derive(Clone, Debug)]
+pub struct DetExpr(Expr);
+
+impl DetExpr {
+    pub fn new(e: Expr) -> Self {
+        Self(e)
+    }
+}
+
+impl WrappedExpr for DetExpr {
+    fn wrap(&self) -> Expr {
+        Wrap::new(self.clone())
+    }
+}
+
+impl ExprSize for DetExpr {
+    fn size(&self) -> usize {
+        self.0.size() + 1
+    }
+
+    fn extraible(&self) -> bool {
+        true
+    }
+
+    fn args(&self) -> Vec<Expr> {
+        vec![self.0.clone()]
+    }
+
+    fn replace_args(&self, args: &[Option<Expr>]) -> Result<Option<Expr>> {
+        Ok(match args {
+            [None] => None,
+            [Some(expr)] => Some(expr),
+            _ => bail!("DetExpr expects 1 argument"),
+        }
+        .map(|expr| -> Expr { Wrap::new(Self(expr.clone())) }))
+    }
+}
+
+impl ConstantFolding for DetExpr {
+    fn as_const(&self) -> Option<Felt> {
+        None
+    }
+
+    fn fold(&self, prime: &Felt) -> Option<Expr> {
+        self.0
+            .fold(prime)
+            .map(|inner| -> Expr { Wrap::new(Self(inner)) })
+    }
+
+    fn replaced_by_const(&self, map: &HashMap<VarStr, Felt>) -> Option<Expr> {
+        self.0
+            .replaced_by_const(map)
+            .map(|inner| -> Expr { Wrap::new(Self(inner)) })
+    }
+}
+
+impl TextRepresentable for DetExpr {
+    fn to_repr(&self) -> TextRepresentation<'_> {
+        owned_list!("det", &self.0)
+    }
+
+    fn width_hint(&self) -> usize {
+        5 + self.0.width_hint()
```

Comment:
Should this be 6 or 7. I haven't run it but I assume it comes out with 3 for "det" plus 2 for start and end brackets and then either 1 or 2 for just a space or comma plus space.

Reply:
Yeah, it should probably be 6